### PR TITLE
Streamline fx mobile grid with one column hidden

### LINKED ISSUE #15648: Heading Text Width on Product Pages
IMG_0028
IMG_0025
Screenshot 2024-12-03 at 4 04 51 PM

On my desktop, there is the input box that lets you send a download link to your phone. On my ipad, however, this input box is not there, but the layout is still two columns, which squishes the text to the left. You can find this happening on the IOS, Android, and Focus pages.

### PR DESCRIPTION
**Pull request by janbrasna**
## One-line summary

When on `.ios` or `.android` the send-to-device form is hidden, making the 2col split with just empty space in 50% of the screen feel like a bug. This overrides the split to a single column to make better use of the space (esp. on mobile).

## Significant changes and points to review

For cases when the form is hidden (making the second column empty), this additionally:
1. resets the grid/display for MD+ breakpoints
2. and also constraints the width in LG+ to be able to keep the noodle graphics but not overflow into it (too much).

## Issue / Bugzilla link

Resolves #15648

## Testing

http://localhost:8000/hi-IN/firefox/browsers/mobile/focus/
http://localhost:8000/ar/firefox/browsers/mobile/android/
http://localhost:8000/el/firefox/browsers/mobile/ios/

(compare desktop vs. mobile device or spoof the uastring to include "android" and "mobile", portrait vs. landscape mode and several breakpoints…)

- - - - -

After:  ×  Before:

Screenshot 2024-12-04 at 1 28 16
Screenshot 2024-12-04 at 1 33 52
Screenshot 2024-12-05 at 13 45 19

### CODE REVIEW

**Comment by maureenlholland** [comment]
might have overlapping changes with another PR: https://github.com/mozilla/bedrock/issues/15648#issuecomment-2516230109

**Comment by janbrasna** [comment]
Yup I'm watching Craig's #15654 — these solve two isolated issues and may complement each other, and should not interfere in any way or conflict code-wise.

(This one is addressing just the empty space as seen above in the screen shots, the other PR changes text size to be closer to the original, as that can also be seen in the images above demonstrating the heading difference between m24 and prod as well.)

File: media/css/firefox/browsers/mobile.scss
Comment (maureenlholland) [inline media/css/firefox/browsers/mobile.scss:164]: **nit, non-blocking**: we might want to be a bit more conservative with avoiding the noodle background where "focus" pages are concerned because it's a darker background color. This was coming up for me on some landscape testing on tablets (around 1080px), minor edge cases

body-overlap

**Review by maureenlholland — APPROVED** [review]
Makes sense to me. Looks good in browserstack testing and some spoofed dev tools testing 🖼️ 

r+wc

File: media/css/firefox/browsers/mobile.scss
Comment (maureenlholland) [reply media/css/firefox/browsers/mobile.scss:164]: hang on, maybe this was an issue with browserstack, I'm not sure that background is supposed to be there given your screenshots

Comment (janbrasna) [reply media/css/firefox/browsers/mobile.scss:164]: I was actually considering removing the noodle graphics completely, if on mobile, thus in one column generally, yet on wide viewport. But that felt like too much design decisions to consider so I kinda cheated using 100% width only when there's no background, and bumped the 50% split to 65% when the background is present, to give the content a bit more space.

So this can:
- either wait for Craig's type changes to land, and update from that to see the impact,
- remove the part trying to cheat around the noodle background presence and keep it at 50%,
- or remove the noodle on mobile completely, no matter what screen size, and keep that in one column still,
- or get creative around applying css tricks/filters to the noodle to make it e.g. lighter(?) just for this specific case to be on the safe side,
- or add a solid white background to the paragraph text in case it runs into the image to keep legibility in worst case?

Comment (maureenlholland) [reply media/css/firefox/browsers/mobile.scss:164]: > either wait for Craig's type changes to land, and update from that to see the impact,
remove the part trying to cheat around the noodle background presence and keep it at 50%,

these options sound best to me 

if the 50% doesn't seem like a generous enough line length on large screens, I would say the background removal is next best option (it's not really serving the purpose of highlighting a call to action anymore and readability takes priority)

Comment (janbrasna) [reply media/css/firefox/browsers/mobile.scss:164]: And of course, with the pending type scale changes it starts to move upwards and collide with the background more: _(so it's def not resilient enough, just running this over the other branch…)_

Screenshot 2024-12-04 at 20 06 17
Screenshot 2024-12-04 at 20 05 44

I'll experiment for a bit just out of curiosity, but eventually once these type changes land I'll rebase out most of it to just fill the empty space.

Comment (janbrasna) [reply media/css/firefox/browsers/mobile.scss]: So now with the type changes in prod I went ahead and did find some "worst case" viewports for this change:

Screenshot 2024-12-05 at 0 05 01
Screenshot 2024-12-05 at 0 04 20
Screenshot 2024-12-04 at 23 50 04
Screenshot 2024-12-04 at 23 48 02
Screenshot 2024-12-04 at 23 51 40
Screenshot 2024-12-04 at 23 52 16

(it's only hypothetical a landscape tablet would hit this exact resolution but not impossible…)

Comment (janbrasna) [reply media/css/firefox/browsers/mobile.scss]: So I went ahead and moved the noodle a few %% out of way, as it serves no purpose in this case without presenting the form, and the same viewport is no longer an issue:

Screenshot 2024-12-05 at 0 02 39
Screenshot 2024-12-05 at 0 03 06
Screenshot 2024-12-05 at 0 03 28

A bit too specific perhaps, but tried;]

Comment (stephaniehobson) [reply media/css/firefox/browsers/mobile.scss]: There's no reason to cut this close. Give those noodles a real boot out of the way 🙂

Comment (janbrasna) [reply media/css/firefox/browsers/mobile.scss]: I think I'm happy now, only keeping a half of every illustration with some low effort styling. Will post some screens (tomorrow) to see if that's viable, or warrants more violent approach O:) — thanks for the feedback!

**Pull request merged**
